### PR TITLE
[WIP] Highlighting selected stream.

### PR DESCRIPTION
**Pull request by plugyawn**
**What does this PR do?**  

In the side panel, once narrowed, the active stream is highlighted.

Partial fix for #516.
Supersedes #1234 

https://chat.zulip.org/#narrow/stream/206-zulip-terminal/topic/Highlight.20currently.20narrowed.20stream.2Ftopic.20.23T516

**Tested?** 
- [x] Manually
- [ ] Existing tests (adapted, if necessary)
- [ ] New tests added (for any new behavior)
- [ ] Passed linting & tests (each commit)

**Commit flow** 

**Notes & Questions** 

**Interactions** 

**Visual changes** 
image

There used to be no visual indicator on the side panel showing which stream was selected.

### CODE REVIEW

**Review by neiljp — COMMENTED** [review]
@plugyawn This almost works for me right now - I added a line to set the style, as we do in update_widget to use set_attr_map, and it performs rather well :+1:

There are a few cases where the previous narrow remains marked, eg. stream->all-messages (press `a` while on a message). In those cases the previous stream stays selected, even though one is in the all-messages narrow. As soon as one selects a different narrow, this is updated though :+1:

While I do think we should consider storing the UI state outside of the model, before we approach that I think it would be beneficial to first get this code into a mergeable set of commits and fix the issue above (and any important others we find along the way). For example, add the new style in a separate commit first.

As part of getting the commits mergeable, also note that the additional style confuses ZT on startup; if you git-grep for existing styles then you'll see there is an existing location where they are specified for mono mode support and also for testing which styles a complete theme should have.

While this PR does interact with @Rohitth007 's #1088, the logic is separate and it's only the style setting that should be affected.

**Comment by neiljp** [comment]
Any feedback on this by others would be great; the line I added was this, to the end of `update_widget_highlight`:
```python3
self._w.set_attr_map({None: highlight_style})

File: zulipterminal/ui_tools/buttons.py [hunk 1]
```diff
@@ -82,6 +83,9 @@ def update_widget(
         self.button_suffix.set_text(suffix)
         self._w.set_attr_map({None: text_color})
 
+    def update_widget_highlight(self, highlight_style: str) -> None:
+        self.original_color = highlight_style
```
Comment (srdeotarse) [inline zulipterminal/ui_tools/buttons.py:87]: In WSL, it did not highlight active stream until I added line ``self._w.set_attr_map({None: highlight_style})`` as suggested by ``Neil``.

**Review by srdeotarse — CHANGES_REQUESTED** [review]
* @plugyawn The selected stream highlight works fine and looks fantastic for ZT but, I also faced the same glitch that selected stream remained highlighted if switched to ``All messages``.
* Also, the selected stream highlight is working as expected when switched to ``PMs``, ``Starred messages`` and ``Mentions``.

Some suggestions to extend this selected item highlighting -
* We can also extend it to ``All messages``, ``PMs``, ``Starred messages`` and ``Mentions`` similar to selected stream highlighting. (This could be accomplished in a different PR)
* Also, we can implement highlighting ``user name`` or ``stream`` if we toggle ``information popup`` using ``i``

**Comment by Rohitth007** [comment]
@plugyawn, apart from the topic panel doubt which I left in the stream:
* There's no need to create the `update_widget_highlight()` function. We can do everything directly from inside `mark_(in)active()`.
* Also don't use `self.original_color` as:
  * It will be removed by #1088 and
  * You would have to go through `update_count()` to use it, especially when we are not actually updating count.

**Comment by neiljp** [comment]
Closing in favor of rebased/readjusted version of later version of this PR, ie. #1334.

**Pull request closed without merging**